### PR TITLE
Remove splitting classes into C (static side) and C_instance

### PR DESCRIPTION
**Pull request by rkirov**
This was needed to support @constructor es5 classes that had custom
semantics in closure. These days most code is using es6 classes, so the
work-around is not needed and it greatly simplifies emitted .d.ts.

### CODE REVIEW

**Comment by rkirov** [comment]
BTW, the commit has a lot better message now, but github does not update the PR message after force pushing the branch. You have to click through to see the commit message.

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java [hunk 1]
```diff
@@ -2586,7 +2525,13 @@ private void visitObjectType(
           ((ObjectType) instanceType).getOwnPropertyNames(),
           superClassFields,
           classTemplateTypeNames);
-      // Statics are handled in INSTANCE_CLASS_SUFFIX class.
+
+      // Interface statics were emitted in an sibling namespace prior to this point, because
```
Comment (mprobst) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: this needs more docs, it describes a situation, but not why the code below has to run.

Also, this method would benefit from breaking it up into sub steps, this is a pretty long read.

**Comment by mprobst** [comment]
@rkirov your commit message should mention that we're intentionally accepting that we produce code that contains TS level errors. Your example also is missing an `@extends`, I believe.

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java (hunk 1 above)
Comment (rkirov) [reply src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: Improved the comment. Will refactor later, as this PR is already on the large side.

**Comment by rkirov** [comment]
"With this change some closure code can produce .d.ts files that are not valid TS." is where I say that. Isn't that clear?

**Comment by mprobst** [comment]
@rkirov sorry, must have missed that. That's clear indeed.

**Pull request merged**